frontend/app/src/scenes/Home/scenes/UpsertCategory/index.tsx

```diff
@@ -50,7 +57,20 @@ class UpsertCategory extends React.Component<
       )
       .then((result: AxiosResponse<Category>) => {
         if (result.status == 200) {
-          this.setState({ isLoading: false, category: result.data });
+          this.setState({
+            isLoading: false,
+            category: result.data,
+          });
+          // Gets the translations for the fetched category and setting them to the state using a map
+          this.state.category?.translations.forEach((translation) => {
+            const categoryTranslations = new Map(this.state.translations);
```

Comment:
Why do you create a new map here?

Reply:
To set the state without directly changing it.

Reply:
create the map object first. Then set the state once. (after doing all the iterations) 

you can set 
isLoading, category, and translations using single setState. 

`setState` is expensive because each time you do, it will rerender the component